examples/mqtt/README.md

```diff
@@ -0,0 +1,80 @@
+
+
+## Credentials and TLS Configurations
+* **Credentials class** encapsulates certificates or credentials related to a connection used at both DCC and Device side.
+* **TLSConf class** encapsulates parameters related to TLS configuration.
+
+![Example](/examples/mqtt/dcc_comms/aws_iot/simulated_home_topic_per_metric.py) shows how the above objects can be used to establish a secure connection.
+
+
+## Using MqttDeviceComms
+
+For both ProjectICE & non-ProjectICE use-cases, MQTT related parameters required in `publish()` and `subscribe()` like topic, qos, etc., are fetched directly from the user's
+configuration file.  Please refer this ![example](/examples/mqtt/iot_cc/iotcc_simulated_mqtt.py) for detailed explanation.
+
+
+## Using MqttDccComms
+
+MqttDccComms provides the flexibility of having a single topic (auto-generated or from configuration file) per MQTT connection or individual topic per metric. **MqttMessagingAttributes class** in ![mqtt.py](/liota/lib/transports/mqtt.py)
+provides this flexibility.
+
+### MqttMessagingAttributes
+
+It enables the following options for developers in LIOTA.
+
+**(a)** Use single publish and subscribe topic generated by LIOTA for an EdgeSystem, its Devices and its Metrics. By default:
+* Publish topic for all Metrics will be **liota/generated_local_uuid_of_edge_system**
+* Subscribe topic will be **liota-resp/generated_local_uuid_of_edge_system**
+* Publish and Subscribe QoS will be **One**
+* Publish retain flag will be **False** and
+* Subscribe call_back will be **None**
+
+**(b)** Use custom single publish and subscribe topic for an EdgeSystem, its Devices and Metrics.  Along with custom topics, other parameters (QoS, retain_flag ,etc.,) can be referred from ![Property File](/examples/mqtt/aws_iot/awsSampleProp.conf).
+
+**-** In the above two cases, MQTT message's payload MUST be self-descriptive so that subscriber can subscribe process accordingly to a single topic by parsing payload.
+
+**(c)** Use custom publish and subscribe topics for Metrics.
+
+**-** In this case, MQTT message's payload need not be self-descriptive.  Subscribers can subscribe to appropriate topics accordingly.
+
+**(d)** Use combination of **(a) and (c)** or **(b) and (c)**.
+
+
+**NOTE:**
+If **Option (a)** is used, `generated_local_uuid_of_edge_system` will be written to a file available at path **uuid_path** as specified in ![liota.conf](/config/liota.conf) file.
+Users can refer this file to get the uuid for subscribing.
+
+
+Configuring MqttDccComms and MessagingAttributes for ProjectICE and non-ProjectICE related use-cases is explained in the following sections.
+
+### Configuration for ProjectICE Scenario
+![IoTCC DCC](/liota/dccs/iotcc.py) will support both WebSockets and MQTT as transports.  **This is WIP**.  Once developed it'll allow plug-and-play approach to easily switch between these two protocols.  However, for MQTT
```

Comment:
In Project ICE, publish topic would be `liota/generated_local_uuid`.  In non-Project ICE it need not be the case.  So, there is separate configuration for non-project ICE. For Project ICE scenarios, it'll remain similar to existing configuration.